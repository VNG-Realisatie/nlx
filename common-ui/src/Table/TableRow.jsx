import styled from 'styled-components'

const TableRow = styled.tr`
    display: table-row;

    &:last-child {
        td, th {
            border-bottom: none;
        }
    }
`

export default TableRow