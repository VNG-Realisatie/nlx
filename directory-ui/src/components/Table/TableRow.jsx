import styled from 'styled-components'

const TableRow = styled.tr`
    display: table-row;
    
    td {
      box-shadow: 0 -1px 0 #F0F2F7;
    
      &:first-child {
        box-shadow: 0 -1px 0 #F0F2F7, -1px 0 #F0F2F7;
      }
      
      &:last-child {
        box-shadow: 0 -1px 0 #F0F2F7, 1px 0 #F0F2F7;
      }
    }
`

export default TableRow
