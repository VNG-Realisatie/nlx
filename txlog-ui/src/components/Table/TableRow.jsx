// Copyright Â© VNG Realisatie 2018
// Licensed under the EUPL

import React, { Component } from 'react'
import PropTypes from 'prop-types'

const selectedData = [
    'created',
    'logrecord-id',
    'source_organization',
    'service_name',
]

class TableRow extends Component {
    render() {
        const { data, code } = this.props
        const time = new Date(data['created']).toLocaleString()
        return (
            <tr className={code && 'code'}>
                {selectedData.map((property, i) => (
                    <td key={i}>
                        {property === 'created'
                            ? time
                            : data[property].toString()}
                    </td>
                ))}
                <td>
                    {Object.keys(data['data']).map((key) => (
                        <span key={key}>
                            <b>{key}</b> {data['data'][key]}
                        </span>
                    ))}
                </td>
            </tr>
        )
    }
}

TableRow.propTypes = {
    data: PropTypes.shape({
        created: PropTypes.string.isRequired,
        'logrecord-id': PropTypes.string.isRequired,
        source_organization: PropTypes.string.isRequired,
        service_name: PropTypes.string.isRequired,
        data: PropTypes.object.isRequired,
    }),
}

export default TableRow
