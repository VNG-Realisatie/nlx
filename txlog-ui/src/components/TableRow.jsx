import React from 'react'

const selectedData = [
    'created',
    'logrecord-id',
    'source_organization',
    'service_name',
]

export default class TableRow extends React.Component {
    render() {
        const { data, code } = this.props

        const time = new Date(data['created']).toLocaleString()

        return (
            <React.Fragment>
                <tr className={code && 'code'}>
                    {selectedData.map((property, i) => (
                        <td key={i}>
                            {property === 'created'
                                ? time
                                : data[property].toString()}
                        </td>
                    ))}
                    <td>
                        {Object.keys(data['data']).map((key) => (
                            <span key={key}>
                                <b>{key}</b> {data['data'][key]}
                            </span>
                        ))}
                    </td>
                </tr>
            </React.Fragment>
        )
    }
}
