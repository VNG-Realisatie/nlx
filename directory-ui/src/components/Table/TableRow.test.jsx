import React from 'react'
import { shallow } from 'enzyme'
import TableRow from './TableRow'

it('should match snapshot', () => {
  const wrapper = shallow(<TableRow/>)
  expect(wrapper).toMatchSnapshot()
})

it('should render child elements', () => {
  const wrapper = shallow(<TableRow><th>Table head</th></TableRow>);
  expect(wrapper.contains(<th>Table head</th>)).toEqual(true)
})