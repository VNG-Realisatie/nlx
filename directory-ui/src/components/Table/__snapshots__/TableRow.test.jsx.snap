// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
<StyledComponent
  forwardedComponent={
    Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "sc-bdVaJa",
        "isStatic": true,
        "rules": Array [
          "
    display: table-row;
    
    td {
      box-shadow: 0 -1px 0 #F0F2F7;
    
      &:first-child {
        box-shadow: 0 -1px 0 #F0F2F7, -1px 0 #F0F2F7;
      }
      
      &:last-child {
        box-shadow: 0 -1px 0 #F0F2F7, 1px 0 #F0F2F7;
      }
    }
",
        ],
      },
      "displayName": "styled.tr",
      "foldedComponentIds": Array [],
      "render": [Function],
      "styledComponentId": "sc-bdVaJa",
      "target": "tr",
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    }
  }
  forwardedRef={null}
/>
`;
